heartbeat/monitors/active/tcp/endpoint.go

```diff
@@ -0,0 +1,108 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package tcp
+
+import (
+	"fmt"
+	"net"
+	"net/url"
+	"strconv"
+
+	"github.com/pkg/errors"
+)
+
+// endpoint configures a host with all port numbers to be monitored by a dialer
+// based job.
+type endpoint struct {
+	Scheme   string
+	Hostname string
+	Ports    []uint16
+}
+
+// perPortURLs returns a list containing one URL per port
+func (e endpoint) perPortURLs() (urls []*url.URL) {
+	for _, port := range e.Ports {
+		urls = append(urls, &url.URL{
+			Scheme: e.Scheme,
+			Host:   net.JoinHostPort(e.Hostname, strconv.Itoa(int(port))),
```

Comment:
Can `Hostname` already have a port number configured?

Reply:
No, the `endpoint` type has had this removed. The goal of `endpoint` is to normalize things so we have a `hostname` instead of a `host` (following the convention of go's URL package where a `Host` is a URL+Port and a `Hostname` is just the name).

So, if there was a port combined with a hostname it's been split out.